src/domain/home/form/HomePreliminaryForm.tsx

```diff
@@ -54,8 +51,15 @@ const HomePreliminaryForm: FunctionComponent<Props> = ({
       verifyInformation: values.verifyInformation,
     });
     setFormValues(payload);
-    if (isAuthenticated) history.push('/registration/form');
-    else loginTunnistamo(`/registration/form`);
+    handleRedirect(payload);
+  };
+
+  const handleRedirect = (payload: RegistrationFormValues) => {
```

Comment:
This look bad for reading.
few if will look better

Reply:
So move back to the previous solution, but still inside a handleRedirect?

Reply:
no, just make the logic less messy to read, keep the function
handleRedirect {
 if(notEligible) {
  goto(somewhere)
}
if(notAuthed) {
  goLoginMate()
}
gotoForm()
}

Reply:
Like this?